### PR TITLE
HDDS-11671. Refer to website for supported versions

### PR DESCRIPTION
**Pull request by xichen01**
## What changes were proposed in this pull request?
Based on the discussion in [the Ozone release rc2 email](https://lists.apache.org/thread/ncpxccgk250g8hwyy6pxhl7v87xjjs68), we should add the ozone-1.4.1 release information to `SECURITY.md` on the ozone-1.4 branch .

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-11671

## How was this patch tested?
https://github.com/xichen01/ozone/actions/runs/11747419785

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ The first stable release of Apache Ozone is 1.0, the previous alpha and beta rel
 | 1.2.1         | :x:                |
 | 1.3.0         | :x:                |
 | 1.4.0         | :white_check_mark: |
```
Comment (adoroszlai) [inline SECURITY.md:17]: Is 1.4.0 still supported after 1.4.1?  How do we release a fix for 1.4.0?

Comment (xichen01) [reply SECURITY.md:17]: So we should only mark the `1.4.1` as "white_check_mark" right?
```suggestion
| 1.4.0         | :x: |
```

Comment (xichen01) [reply SECURITY.md:17]: Or we should delete this `SECURITY.md` file now, and put it to the website, in the future.

Comment (adoroszlai) [reply SECURITY.md:17]: https://github.com/apache/ozone-site/pull/102

Comment (adoroszlai) [reply SECURITY.md:17]: I think we should still keep `SECURITY.md`, but replace the `Supported Versions` table with a link to the website.

File: SECURITY.md [hunk 2]
```diff
@@ -2,19 +2,7 @@
...
-| 0.3.0 (alpha) | :x:                |
-| 0.4.0 (alpha) | :x:                |
-| 0.4.1 (alpha) | :x:                |
-| 0.5.0 (beta)  | :x:                |
-| 1.0.0         | :x:                |
-| 1.1.0         | :x:                |
-| 1.2.1         | :x:                |
-| 1.3.0         | :x:                |
-| 1.4.0         | :white_check_mark: |
+Supported versions of Apache Ozone can be found on the [Apache Ozone Downloads](https://ozone.apache.org/downloads/).
```
Comment (adoroszlai) [inline SECURITY.md:5]: I would rephrase it a bit:

```suggestion
Please check the Apache Ozone [website](https://ozone.apache.org/downloads/) for the list of versions currently supported.
```

but further suggestions are welcome.

Comment (xichen01) [reply SECURITY.md]: I think this change is OK.

**Review by adoroszlai — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @xichen01 for the update.
